### PR TITLE
new blob file iterator

### PR DESCRIPTION
**Pull request by wjhuang2016**

### CODE REVIEW

File: src/blob_file_iterator.h [hunk 1]
```diff
@@ -33,6 +34,9 @@ class BlobFileIterator {
   Status status() const { return status_; }
 
   void IterateForPrev(uint64_t);
+  Status PunchHole(uint64_t offset, size_t n) {
```
Comment (lhy1024) [inline src/blob_file_iterator.h]: It is not recommended to define functions directly in the header file.

**Review by lhy1024 — COMMENTED** [review]
Some formats still have unnecessary modifications. Do we need to save them ?

File: src/blob_file_iterator.h (hunk 1 above)
Comment (wjhuang2016) [reply src/blob_file_iterator.h]: Done

**Comment by wjhuang2016** [comment]
@lhy1024 PTAL

**Review by lhy1024 — APPROVED** [review]
LGTM

**Pull request merged**